app/tests/test_sessions.py

```diff
@@ -46,15 +46,16 @@ def test_create_session_with_valid_quiz_id_and_first_session(self):
         data = open("app/tests/dummy_data/homework_quiz.json")
         quiz_data = json.load(data)
         response = self.client.post(quizzes.router.prefix + "/", json=quiz_data)
-        quiz = json.loads(response.content)
+        quiz_id = json.loads(response.content)["quiz_id"]
+        quiz = self.client.get(quizzes.router.prefix + f"/{quiz_id}").json()
         response = self.client.post(
             sessions.router.prefix + "/", json={"quiz_id": quiz["_id"], "user_id": 1}
         )
         assert response.status_code == 201
         session = json.loads(response.content)
         assert session["is_first"] is True
-        assert len(session["session_answers"]) == len(
-            quiz_data["question_sets"][0]["questions"]
+        assert len(session["session_answers"]) == sum(
```

Comment:
What is this change for?

Reply:
previous code assumes that there is only one question set, once we include optional we'll have more than one question set, so I edited the line to iterate over all question sets (qset)

Reply:
Ah shouldn't be a part of this PR then -- but is one line so I guess it's okay. Anyway, that's why I got confused.